src/bindings/xfconf/elektra-xfconf.c

```diff
@@ -0,0 +1,73 @@
+#include "elektra-xfconf.h"
+#include "elektra-xfconf-util.h"
+
+GList * channel_list = NULL;
+KDB * elektraKdb = NULL;
+
+gboolean xfconf_init (GError ** error)
+{
+	trace ();
+	Key * elektraError = keyNew ("/elektra_error", KEY_END);
+	elektraKdb = kdbOpen (NULL, elektraError);
+	if (elektraKdb == NULL)
+	{
+		g_debug ("unable to open elektraKdb");
+		*error = g_error_new (0, 1, "unable to open kdb: %s", keyString (elektraError));
+		return FALSE;
+	}
+	return TRUE;
+}
+void xfconf_shutdown (void)
+{
+	trace ();
+	g_list_free (channel_list);
+}
+
+void xfconf_named_struct_register (const gchar * struct_name, guint n_members, const GType * member_types)
+{
+	unimplemented ();
+}
+
+void xfconf_array_free (GPtrArray * arr)
+{
+	unimplemented ();
+}
+
+gchar ** xfconf_list_channels (void)
+{
+	trace ();
+	Key * parentKey = keyNew (XFCONF_ROOT, KEY_END);
+	KeySet * channelKeySet = ksNew (0, KS_END);
+	kdbGet (elektraKdb, channelKeySet, parentKey);
+	ssize_t keySetLength = ksGetSize (channelKeySet);
+	gchar ** channelNames = calloc (keySetLength + 1, sizeof (gchar *));
+	const Key * currentKey;
+	const char * currentKeyName;
+	const char * currentChannelNameWithSuffix;
+	const char * currentChannelNameEnd;
+	for (elektraCursor i = 0, nameIndex = 0; i < keySetLength; i++)
+	{
+		currentKey = ksAtCursor (channelKeySet, i);
+		currentKeyName = keyName (currentKey);
+		currentChannelNameWithSuffix = &currentKeyName[strlen (XFCONF_NAMESPACE) + strlen (XFCONF_ROOT) + 1];
+		currentChannelNameEnd = strchr (currentChannelNameWithSuffix, '/');
+		if (!currentChannelNameEnd)
+		{
+			currentChannelNameEnd = &currentChannelNameWithSuffix[strlen (currentChannelNameWithSuffix)];
+		}
+		gchar * firstLevelName = strndup (currentChannelNameWithSuffix, currentChannelNameEnd - currentChannelNameWithSuffix);
```

Comment:
I'm not sure what you're trying to do here, but you should probably use unescaped names (or the `key*Name*` APIs). You're not handling any of the various escapes that might appear in the escaped name.

Reply:
Please add this as comment here.

Reply:
Done with b8c304e21c2aa120e38930bf81ef1c377f96d563.